service.subtitles.ktuvit/service.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+
+from urllib.parse import unquote
+
+__addon__ = xbmcaddon.Addon()
+__author__ = __addon__.getAddonInfo('author')
+__scriptid__ = __addon__.getAddonInfo('id')
+__scriptname__ = __addon__.getAddonInfo('name')
+__version__ = __addon__.getAddonInfo('version')
+__language__ = __addon__.getLocalizedString
+
+__cwd__ = xbmcvfs.translatePath(__addon__.getAddonInfo('path'))
+__profile__ = xbmcvfs.translatePath(__addon__.getAddonInfo('profile'))
+__resource__ = xbmcvfs.translatePath(os.path.join(__cwd__, 'resources', 'lib'))
+__temp__ = xbmcvfs.translatePath(os.path.join(__profile__, 'temp'))
+
+sys.path.append(__resource__)
+
+from SUBUtilities import SubsHelper, log, normalizeString, parse_rls_title, clean_title
+
+
+def search(item):
+    helper = SubsHelper()
+    subtitles_list = helper.get_subtitle_list(item)
+    if subtitles_list:
+        for it in subtitles_list:
+            listitem = xbmcgui.ListItem(label=it["language_name"], label2=it["filename"])
+            listitem.setArt({'icon': str(it["rating"]), 'thumb': it["language_flag"]})
+
+            if it["sync"]:
+                listitem.setProperty("sync", "true")
+            else:
+                listitem.setProperty("sync", "false")
+
+            if it.get("hearing_imp", False):
+                listitem.setProperty("hearing_imp", "true")
+            else:
+                listitem.setProperty("hearing_imp", "false")
+
+            url = "plugin://%s/?action=download&id=%s&sub_id=%s&filename=%s&language=%s" % (
+                __scriptid__, it["id"], it["sub_id"], it["filename"], it["language_flag"])
+            xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=listitem, isFolder=False)
+
+
+def download(id, sub_id, filename, language):
+    subtitle_list = []
+    exts = [".srt", ".sub"]
+
+    filename = os.path.join(__temp__, "%s.%s.srt" % (filename, language))
+
+    helper = SubsHelper()
+    helper.download(id, sub_id, filename)
+
+    for file in xbmcvfs.listdir(__temp__)[1]:
+        full_path = os.path.join(__temp__, file)
+        if os.path.splitext(full_path)[1] in exts:
+            subtitle_list.append(full_path)
+
+    return subtitle_list
+
+
+def get_params(string=""):
```

Comment:
This this ugly as hell function can be replaced with `dict(parse_qsl(sys.argv[2][1:]))`